File: ad.cpp
#include "advertiser.cpp"

class Ad: public BaseAdvertising{
    public:
        Ad(int _id, string _title, string _imgUrl, string _link, Advertiser *_advertiser){
            id = _id;
            title = _title;
            imgUrl = _imgUrl;
            link = _link;
            views = 0;
            clicks = 0;
            setAdvertiser(_advertiser);
        }
        string getTitle(){return title;}
        string getImgUrl(){return imgUrl;}
        void setImgUrl(string url){imgUrl = url;}
        string getLink(){return link;}
        void setLink(string _link){link = _link;}
        void setAdvertiser(Advertiser *_advertiser){ advertiser = _advertiser;}
        string describeMe(){return "Ad class stores information of advertisement.";}
        void incClicks() {
            clicks += 1;
            advertiser->incClicks();
            }
        void incViews() {
            views += 1;
            advertiser->incViews();
            }


    private:
        int id;
        string title;
        string imgUrl;
        string link;
        Advertiser *advertiser;

};

File: byPython/ad.py
import advertiser
import baes_model

class Ad(baes_model.BaseAdvertising):

    def __init__(self, _id, _title, _imgUrl, _link, _advertiser):
        self.__id = _id
        self.__title = _title
        self.__imgUrl = _imgUrl
        self.__link = _link
        self.views = 0
        self.clicks = 0
        self.setAdvertiser(_advertiser)

    def getTitle(self): 
        return self.__title

    def getImgUrl(self):
        return self.__imgUrl

    def setImgUrl(self, url):
        self.__imgUrl = url

    def getLink(self):
        return self.__link

    def setLink(self, _link):
        self.__link = _link

    def setAdvertiser(self, _advertiser):
        self.__advertiser = _advertiser

    def describeMe(self):
        return "Ad class stores information of advertisement."

    def incClicks(self):
            self.clicks += 1
            self.__advertiser.incClicks()

    def incViews(self):
            self.views += 1
            self.__advertiser.incViews()
